### PR TITLE
Card and elevation implementation

### PR DESCRIPTION
**Pull request by rbeard0330**
Implements Card, with doc example.

The main difference from the Blueprintjs version is that the `Elevation` enum is more feature-ful: it can be converted into a number, and `u8`s can be converted into a variant of the enum.

I look forward to getting comments.

### CODE REVIEW

**Comment by cecton** [comment]
You should use yew's Classes instead of string. You can probably get rid of the whole impl with that

**Comment by cecton** [comment]
I think you can delete all conversion to u8. It's not like we're going to make arithmetic operations on it. Try this instead,

```rs
fn up(&self) -> Elevation {
    match self {
        Elevation0 => Elevation1,
        Elevation1 => Elevation2,
...
        Elevation5 => Elevation5,
    }
}
```

Same for `down()` probably.

**Comment by cecton** [comment]
You can also move Elevation to lib.rs because I think it is used elsewhere. But don't associate a numeric value to the enum imo, it isn't really numeric, it's api should be very simple to use

**Comment by rbeard0330** [comment]
OK, will make those changes.  Do you want to keep the `value()` method or drop that as well?

**Comment by cecton** [comment]
You COULD if you want make 4 functions, saturated_up, checked_up, saturated_down, checked_down if you see what I mean because I suspect it could be useful but I'm not sure.

I will merge tomorrow. Don't worry for the merge conflicts.

It's really excellent code. It's a very fine first component, congrats! :heart:

**Comment by cecton** [comment]
No I think you can drop value too. This shouldn't be used as a state.

**Comment by rbeard0330** [comment]
> You COULD if you want make 4 functions, saturated_up, checked_up, saturated_down, checked_down if you see what I mean because I suspect it could be useful but I'm not sure.
> 
> I will merge tomorrow. Don't worry for the merge conflicts.
> 
> It's really excellent code. It's a very fine first component, congrats! ❤️

Thanks for the kind words!

**Comment by cecton** [comment]
Ah thanks you updated the branch and resolved the conflicts properly!

I can't modify your branch because it is on your own fork. Next time feel free to make the branch directly on cecton/yewprint if you want. You have access.

CI is failing because you didn't run `cargo fmt`. The rest seems to pass :+1:

File: src/elevation.rs [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    pub (crate) fn as_css_class(&self) -> &'static str {
+        use Elevation::*;
+        match self {
+            Level0 => "bp3-elevation-0",
+            Level1 => "bp3-elevation-1",
+            Level2 => "bp3-elevation-2",
+            Level3 => "bp3-elevation-3",
+            Level4 => "bp3-elevation-4",
+        }
+    }
```
Comment (cecton) [inline src/elevation.rs:56]: You can impl a `From` (and `AsRef`) instead like I did here: https://github.com/cecton/yewprint/blob/main/src/lib.rs#L55-L70

With this, it makes it possible to use it in classes directly:

```rs
classes.push(elevation);
```

https://docs.rs/yew/0.17.3/yew/virtual_dom/struct.Classes.html#impl-From%3C%26%27_%20str%3E

File: src/card/mod.rs [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+        html! {
+            <div class=class onclick={self.props.onclick.clone()}>
+                {self.props.children.clone()}
+            </div>
+        }
+    }
+}
+
+#[cfg(feature = "doc")]
+pub mod doc {
```
Comment (cecton) [inline src/card/mod.rs:60]: We just changed it everywhere. Doc now goes to doc.rs (of the component). See #11

**Comment by rbeard0330** [comment]
> Ah thanks you updated the branch and resolved the conflicts properly!
> 
> I can't modify your branch because it is on your own fork. Next time feel free to make the branch directly on cecton/yewprint if you want. You have access.
> 
> CI is failing because you didn't run `cargo fmt`. The rest seems to pass 👍

Got it.  I opened a new branch here, thinking I could just swap it into this PR, but now I see that you can't do it that way... I'm going to close this one and do a PR for the new branch.  Sorry for the confusion--I don't have much/any experience with this stuff.  The new branch has the `From`/`AsRef` traits implemented for `Elevation` and has the doc example in doc.rs

**Pull request closed without merging**